tests/annotation/custom-grant/src/main/java/demo/Application.java

```diff
@@ -87,7 +87,10 @@ public void configure(ClientDetailsServiceConfigurer clients) throws Exception {
 		            .authorities("ROLE_CLIENT")
 		            .scopes("read")
 		            .resourceIds("oauth2-resource")
-		            .secret("secret");
+		            .secret("secret")
+		    .and()
+		        .withClient("my-client-with-unlimited-scopes")
```

Comment:
There are quite a few configurations where this new client is added. Are you able to just add this new client in 1 or 2 configurations only?

Reply:
I added the new test to `AbstractRefreshTokenSupportTests` and hence had to update all 17 subclasses. I can move it to one of the concrete `RefreshTokenSupportTests` instead -- which is canonical?

Reply:
On second look, I don't think we need to add the new client `my-client-with-empty-scopes`. Instead, we can reuse the existing client `my-trusted-client` for the `refresh_token` flow and simply not pass any `scope` parameter. This will test out the new behaviour. Makes sense?

Reply:
Unfortunately not, this issue only occurs when the client is defined with empty scopes. See [my comment](https://github.com/spring-projects/spring-security-oauth/issues/1416#issuecomment-497938510) for a fuller explanation.